Makefile

```diff
@@ -1,4 +1,16 @@
+ifeq (, $(shell which python ))
+  $(error "PYTHON=$(PYTHON) not found in $(PATH)")
+endif
+
 PYTHON = python
+PYTHON_VERSION_MIN=3.6
```

Comment:
I'd only check here for python2. And if it's not python 2 I'd then rely on https://packaging.python.org/guides/distributing-packages-using-setuptools/#python-requires in the setup.cfg, rather than checking it here. That way all the metadata is in one place (going under the assumption people have a updated pip).